python/src/gz/sim/Gui.cc

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <pybind11/pybind11.h>
+
+#include <gz/sim/gui/Gui.hh>
+#include <gz/common/Console.hh>
+
+#include "Server.hh"
+
+namespace gz
+{
+namespace sim
+{
+namespace python
+{
+void defineGuiClient(pybind11::module &_module)
+{
+
+  _module.def("run_gui", [](){
```

Comment:
This is currently a blocker: I'm not sure what the best way forward is.
- Gazebo GUI does not like running in the same process as the server. It core dumps when I thry to run it from within a `std::thread`.
- Forking works on linux, but windows has no concept of `fork()`. It seems to start a new process, you must call an executable via `CreateProcess` instead of simply `fork()`ing.
- An alternative would be to create a jupyter widget out of `gz-web` but that will need us to work on the websocket server.

Reply:
> Forking works on linux, but windows has no concept of fork(). It seems to start a new process, you must call an executable via CreateProcess instead of simply fork()ing.

We had similar problems on Gazebo Classic, in my experience to abstract away the problem of creating processes on Windows it is more convenient with libraries like https://github.com/DaanDeMeyer/reproc or https://gitlab.com/eidheim/tiny-process-library .

Reply:
By the way, I guess we will have a similar problem when implementing `gz-sim` standalone executable for https://github.com/gazebosim/gz-tools/issues/7 .

Reply:
So last night I had a discussion with @azeey. I think the proposed method is to use the `runGuiMain.cc` executable on both platforms. I haven't tried it yet.

Reply:
An update is python's `os.fork()` works via Cygwin on windows and may be a viable option for us.

Reply:
> An update is python's `os.fork()` works via Cygwin on windows and may be a viable option for us.

I am  not sure about that. The official Python binary on Windows from python.org, the python installed by uv and the python installed by conda-forge all use msvc-based python. To use the cygwin powered python, you need to install python via cygwin, and that also prevents you to install most existing Windows wheels, that assume the use msvc-calling convention or runtime libraries (see https://github.com/pypa/cibuildwheel/issues/329).

Reply:
Yeah, I'm going ahaed and using the binary method for now.